dvc/api.py

```diff
@@ -1,11 +1,17 @@
 import os
 from contextlib import _GeneratorContextManager as GCM
+from contextlib import contextmanager
+from typing import Iterator
 
-from funcy import reraise
-
-from dvc.exceptions import OutputNotFoundError, PathMissingError
-from dvc.path_info import PathInfo
+from dvc.exceptions import NoOutputInExternalRepoError, OutputNotFoundError
 from dvc.repo import Repo
+from dvc.repo_path import RepoPath
+
+
+@contextmanager
+def files(path=os.curdir, repo=None, rev=None) -> Iterator[RepoPath]:
```

Comment:
I don't have a better suggestion but IMO, the name `files` doesn't feel very intuitive (or related with what it returns).

Reply:
@daavoo `walk_files`?

Reply:
`files()`/`walk_files` sound to me like what I would expect to be returned by `os.listdir("root")`/`os.walk("root")`. 

However, it looks like what `files()` is returning is more similar to what `Path("root")` would return (unless I'm completely misunderstanding the example snippets in the description).

So it would make more sense a name that doesn't suggest that you are going to return an iterator of files instead of a context manager with a Path-like instance pointing to the root.

Does that make sense?

Reply:
I agree that the usage wasn't immediately intuitive to me. How much value is the 2-line `files()` function adding? It might be more clear to use `Repo.open()` and `RepoPath` directly since they follow conventions of Python built-ins.

Reply:
I am open for changing the name of the API, would love to hear other alternative namings. Regarding `Repo`, it's an undocumented API and to replace `files()`, it needs to interact with two different components.

Reply:
`get_repo_path`?